.github/workflows/check-dependencies.yaml

```diff
@@ -26,17 +30,22 @@ jobs:
       #          install poetry
       #----------------------------------------------
       - name: Install Poetry
-        # Pin to 1.3.2 to workaround https://github.com/python-poetry/poetry/issues/7611
-        run: pipx install poetry==1.3.2
+        run: |
+          pipx install poetry
+          pipx inject poetry poetry-dynamic-versioning
```

Comment:
don't need the dynamic versioning plugin unless you're releasing a package to pypi

Reply:
Your comment made me look at the action in detail: I believe it does not do what it should do. My understanding is that it should run the tests with updated dependencies but it just installs from the current (possibly outdated) lock file. Then it removes the lockfile but does not create an updated one before running the tests. So the tests are exactly the same as when run from main.yaml.

Would you agree that the idea here is to update the lock file first and then test? In this case dynamic-versioning would be needed for the dependency update.

I'll update the logic.

Reply:
@sierra-moxon created the action, but accidentally removed the re-installation with updated dependencies in https://github.com/linkml/linkml-runtime/commit/05792437e27fbdd4fec5aaa64c71100af88a1bc5

Reply:
I also noticed that this action seems to duplicate a lot of the `main.yaml` actions so it may be time for a general GHA overhaul. That's probably an action for a different PR, though.